src/peerd/runtime.rs

```diff
@@ -127,36 +180,45 @@ impl esb::Handler<ServiceBus> for BridgeHandler {
     }
 }
 
-pub struct ListenerRuntime {
+// PeerReceiverRuntime handles incoming messages only
+pub struct PeerReceiverRuntime {
     internal_identity: ServiceId,
     bridge: esb::Controller<ServiceBus, Request, BridgeHandler>,
+    awaiting_pong: bool,
+    _thread_flag_rx: std::sync::mpsc::Receiver<()>,
 }
 
-impl ListenerRuntime {
+impl PeerReceiverRuntime {
     /// send msgs over bridge from remote to local runtime
     fn send_over_bridge(
         &mut self,
         req: <Unmarshaller<Msg> as Unmarshall>::Data,
     ) -> Result<(), Error> {
         debug!("Forwarding FWP message over BRIDGE interface to the runtime");
-        self.bridge.send_to(
+        if let Err(err) = self.bridge.send_to(
             ServiceBus::Bridge,
             self.internal_identity.clone(),
             Request::Protocol((&*req).clone()),
-        )?;
-        Ok(())
+        ) {
+            error!("Error sending over bridge: {}", err);
+            Err(err.into())
+        } else {
+            Ok(())
+        }
     }
 }
 
-use std::fmt::{Debug, Display};
-impl peer::Handler<Msg> for ListenerRuntime {
+impl peer::Handler<Msg> for PeerReceiverRuntime {
     type Error = crate::Error;
     fn handle(
         &mut self,
         message: <Unmarshaller<Msg> as Unmarshall>::Data,
     ) -> Result<(), Self::Error> {
         // Forwarding all received messages to the runtime
         trace!("FWP message details: {:?}", message);
+        if let Msg::Pong(_) = *Arc::clone(&message) {
+            self.awaiting_pong = false;
```

Comment:
if self.awaiting_pong == true { self.awaiting_pong = false } else { error!("unexpected pong") }

Reply:
Done in f222148d42672d34f05f3657aa1e3c2387b0930a